### PR TITLE
Adding metrics:

### PR DESCRIPTION
**Pull request by arcturus**
- Installation
- Launch
- Expand view
- Swipe away

### CODE REVIEW

**Comment by arcturus** [comment]
@wilsonpage feedback? or r? if looks ok.

The big chunk in metrics that is inside a wrapper comes from:

https://github.com/tamarahills/metrics_controller/blob/master/js/metrics.js

Anything else we should add to metrics?

File: android/app/src/main/java/org/mozilla/magnet/MainActivity.java [hunk 1]
```diff
@@ -34,6 +36,7 @@ protected boolean getUseDeveloperSupport() {
     @Override
     protected List<ReactPackage> getPackages() {
         return Arrays.<ReactPackage>asList(
+            new RNDeviceInfo(),
             new MainReactPackage(),
                 new MyAppPackage()
```
Comment (wilsonpage) [inline android/app/src/main/java/org/mozilla/magnet/MainActivity.java:41]: NIT Indentation

File: lib/utils/metrics.js [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+    then(v => {
+      if (v) {
+        return Promise.resolve(true);
+      } else {
+        return this.storage.setItem(KEY, 'true').
+          then(() => {
+            return false;
+          });
+      }
+    }).
```
Comment (wilsonpage) [inline lib/utils/metrics.js]: NIt: dot on next line

File: lib/utils/metrics.js [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+};
+
+/**
+ * Utility function to record an event just once.
+ * For example an installation event, it stores a key in AsyncStorage or other
+ * storage and verify it before registering the event.
+ */
+MagnetMetrics.prototype._recordOnce = function(category, action, label, value) {
+  const KEY = `${category}_${action}_${label}`;
+  return this.storage.getItem(KEY).
```
Comment (wilsonpage) [inline lib/utils/metrics.js]: As below

File: lib/views/app.js [hunk 4]
```diff
@@ -43,6 +44,10 @@ class AppView extends React.Component {
   constructor(props) {
     super(props);
 
+    // Metrics
+    Metrics.recordLaunch();
+    Metrics.recordInstall();
```
Comment (wilsonpage) [inline lib/views/app.js]: Metrics.launch();
Metrics.install();

Metrics.record.launch();
Metrics.record.install();

File: lib/views/list-item/index.js [hunk 5]
```diff
@@ -199,7 +200,10 @@ class ListItem extends React.Component {
     debug('gesture end', vx, exceededThreshold);
 
     // fast swipes fling the tile away
-    if (exceededThreshold) this.swipeAway(vx)
+    if (exceededThreshold) {
+      this.swipeAway(vx);
+      Metrics.recordSwiped();
```
Comment (wilsonpage) [inline lib/views/list-item/index.js]: Move to `.swipeAway()` method.

File: lib/views/list-item/index.js [hunk 6]
```diff
@@ -219,6 +223,9 @@ class ListItem extends React.Component {
     // bubble the event up the tree when not expanded
     // to allow accestors to handle the event
     if (this.isEmbed() || !this.isExpanded()) {
+      if (!this.isExpanded()) {
+        Metrics.recordExpanded();
```
Comment (wilsonpage) [inline lib/views/list-item/index.js]: Move to `ListView.prototype.expand()`

**Comment by arcturus** [comment]
@wilsonpage suggestions added r?

**Comment by wilsonpage** [comment]
@arcturus you need to update the path `Cannot find module '../../utils/metrics'` inside views/list.js

**Comment by arcturus** [comment]
Damn it, thousand of sorries, copy and paste and didn't try that change.

Shame on me.

**Comment by arcturus** [comment]
Fixes #33

**Pull request merged**